hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -476,6 +488,51 @@ private void handleUserTransaction(
         blockRecordManager.endUserTransaction(recordListResult.records().stream(), state);
     }
 
+    /**
+     * Updates key on the hollow accounts that need to be finalized. This is done by dispatching a preceding
+     * synthetic update transaction. The ksy is derived from the signature expansion, by looking up the ECDSA key
+     * for the alias.
+     *
+     * @param context the handle context
+     * @param configuration the configuration
+     * @param accounts the set of hollow accounts that need to be finalized
+     * @param verifier the key verifier
+     */
+    private void finalizeHollowAccounts(
+            @NonNull final HandleContext context,
+            @NonNull final Configuration configuration,
+            @NonNull final Set<Account> accounts,
+            @NonNull final DefaultKeyVerifier verifier) {
+        final var consensusConfig = configuration.getConfigData(ConsensusConfig.class);
+        final var precedingHollowAccountRecords = accounts.size();
+        final var maxRecords = consensusConfig.handleMaxPrecedingRecords();
+        // If the hollow accounts that need to be finalized is greater than the max preceding
+        // records allowed throw an exception
+        if (precedingHollowAccountRecords >= maxRecords) {
+            throw new HandleException(MAX_CHILD_RECORDS_EXCEEDED);
+        } else {
+            for (final var hollowAccount : accounts) {
+                // get the verified key for this hollow account
+                final var verification = verifier.verificationFor(hollowAccount.alias());
+                if (verification.key() != null) {
```

Comment:
Wouldn't it be a catastrophic invariant failure if we got this far with a required hollow account having `null` verification?

Imo we can replace line 516 with,
```
final var verification = Objects.requireNonNull(verifier.verificationFor(hollowAccount.alias()), "Required hollow account verified signature did not exist");
```
...because in any case this should be impossible?

Reply:
Agreed. Changed as suggested